node/cdn/src/blocks.rs

```diff
@@ -228,6 +231,13 @@ pub async fn load_blocks<N: Network>(
         .map_err(|e| (current_height, e))?;
     }
 
+    // While the node will never request the genesis block, non-node callers may do so; in that case 1
+    // needs to be subtracted from the returned value, as the number of processed blocks is lower than
+    // the height of the ledger.
+    if start_height == 0 {
+        current_height = current_height.saturating_sub(1);
+    }
```

Comment:
Why would this impact the current_height? Sorry, I'm not following the change.

Reply:
I can see how this is unclear; I updated the PR and now, instead of any special-casing, the `current_height` just gets updated with the last block's height, which is simpler to digest